sdk/core/Azure.Core/tests/LruCacheTests.cs

```diff
@@ -1,9 +1,10 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
 using NUnit.Framework;
+using Azure.Core;
 
-namespace Microsoft.Azure.Data.SchemaRegistry.ApacheAvro.Tests
+namespace Azure.Core.Tests
 {
     public class LruCacheTests
```

Comment:
I may also have to defer back to @JoshLove-msft for this one. I personally haven't tested the thread-safety of this cache. From my understanding, since all of the code in `TryGet` and `AddOrUpdate` is contained within a lock on `_syncLock`, this will provide exclusive thread access to the linked list, the dictionary map, and allows each thread to do all of its checks and updates before the next thread can start. This should be enough to make this class thread-safe.
It would be straightforward to do some local testing, but I'm not sure how to work thread-safety testing into unit/live tests, it'd probably have to be some kind of stress test?

Reply:
The Count property and the enumeration are not thread safe. I am not sure it matters.